nncf/torch/utils.py

```diff
@@ -132,7 +132,7 @@ def fp32_accum_wrapper(func):
     def wrapper(tensor_to_sum, ret_tensor):
         half = tensor_to_sum.dtype == np.float16
         if half:
-            tensor_to_sum = tensor_to_sum.astype(np.float)
+            tensor_to_sum = tensor_to_sum.astype(np.float64)
```

Comment:
I think it should be cast to float32 if the input type if float32. The same logic as for float16.

Reply:
Change to np.float32